lib/Extractor/Candidates.cpp

```diff
@@ -847,6 +871,64 @@ void ExtractExprCandidates(Function &F, const LoopInfo *LI, DemandedBits *DB,
   for (auto &BB : F) {
     std::unique_ptr<BlockCandidateSet> BCS(new BlockCandidateSet);
     for (auto &I : BB) {
+      if (PrintNegAtReturn && isa<ReturnInst>(I)) {
+        auto V = I.getOperand(0);
+        auto DL = F.getParent()->getDataLayout();
+        bool Negative;
+        Negative = isKnownNegative(V, DL);
+        if (Negative)
+          llvm::outs() << "known negative from compiler: " << "true" << "\n";
+        else
+          llvm::outs() << "known negative from compiler: " << "false" << "\n";
+      }
+      if (PrintNonNegAtReturn && isa<ReturnInst>(I)) {
+        auto V = I.getOperand(0);
+        auto DL = F.getParent()->getDataLayout();
+        bool NonNegative;
+        NonNegative = isKnownNonNegative(V, DL);
+        if (NonNegative)
+          llvm::outs() << "known nonNegative from compiler: " << "true" << "\n";
+        else
+          llvm::outs() << "known nonNegative from compiler: " << "false" << "\n";
+      }
+      if (PrintKnownAtReturn && isa<ReturnInst>(I)) {
+        auto V = I.getOperand(0);
+        auto DL = F.getParent()->getDataLayout();
+        unsigned Width = DL.getTypeSizeInBits(V->getType());
+        KnownBits Known(Width);
+        computeKnownBits(V, Known, DL);
+        llvm::outs() << "knownBits from compiler: " << Inst::getKnownBitsString(Known.Zero, Known.One) << "\n";
+      }
+      if (PrintPowerTwoAtReturn && isa<ReturnInst>(I)) {
+        auto V = I.getOperand(0);
+        auto DL = F.getParent()->getDataLayout();
+        bool PowerTwo;
+        PowerTwo = isKnownToBeAPowerOfTwo(V, DL);
+        if (PowerTwo)
+          llvm::outs() << "known powerOfTwo from compiler: " << "true" << "\n";
+        else
+          llvm::outs() << "known powerOfTwo from compiler: " << "false" << "\n";
+      }
+      if (PrintNonZeroAtReturn && isa<ReturnInst>(I)) {
```

Comment:
instead of these repeated tests of ```isa<ReturnInst>``` just test this once at the beginning and then test the individual flags